packages/netlify-cms-core/src/actions/editorialWorkflow.js

```diff
@@ -310,7 +310,7 @@ export function persistUnpublishedEntry(collection, existingUnpublishedEntry) {
      * Serialize the values of any fields with registered serializers, and
      * update the entry and entryDraft with the serialized values.
      */
-    const fields = selectFields(collection, entry.get('slug'));
+    const fields = selectFields(collection, entry.get('slug'), entry);
```

Comment:
We should test that this works with and without the editorial workflow - as it stands it would only impact projects using the workflow. Actions for non-workflow projects are in `actions/entries.js`.

Reply:
I'm going to work on getting tests put together for the ```persistEntry``` and ```persistUnpublishedEntry``` functions. This should cover the two use cases you mentioned and test the outputs for your other suggestion. Pushing up what I have so far.